bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Lock.java

```diff
@@ -39,30 +46,99 @@ static class ChannelConfiguration extends AbstractChannelConfiguration {
 
         protected boolean optimistic = false;
 
+        @SerializedName("command_topic")
+        protected @Nullable String commandTopic;
         @SerializedName("state_topic")
         protected String stateTopic = "";
         @SerializedName("payload_lock")
         protected String payloadLock = "LOCK";
         @SerializedName("payload_unlock")
         protected String payloadUnlock = "UNLOCK";
-        @SerializedName("command_topic")
-        protected @Nullable String commandTopic;
+        @SerializedName("payload_open")
+        protected @Nullable String payloadOpen;
+        @SerializedName("state_jammed")
+        protected String stateJammed = "JAMMED";
+        @SerializedName("state_locked")
+        protected String stateLocked = "LOCKED";
+        @SerializedName("state_locking")
+        protected String stateLocking = "LOCKING";
+        @SerializedName("state_unlocked")
+        protected String stateUnlocked = "UNLOCKED";
+        @SerializedName("state_unlocking")
+        protected String stateUnlocking = "UNLOCKING";
     }
 
+    private boolean optimistic = false;
+    private OnOffValue lockValue;
+    private TextValue stateValue;
+
     public Lock(ComponentFactory.ComponentConfiguration componentConfiguration) {
         super(componentConfiguration, ChannelConfiguration.class);
 
-        // We do not support all HomeAssistant quirks
-        if (channelConfiguration.optimistic && !channelConfiguration.stateTopic.isBlank()) {
-            throw new ConfigurationException("Component:Lock does not support forced optimistic mode");
-        }
+        this.optimistic = channelConfiguration.optimistic || channelConfiguration.stateTopic.isBlank();
+
+        lockValue = new OnOffValue(new String[] { channelConfiguration.stateLocked },
```

Comment:
I could see an argument that UNLOCKING should be considered locked for this binary channel, but for now I took the conservative approach that anything that's not locked should be considered unlocked.